# Lesson 01

1. Из колоды в 52 карты извлекаются случайным образом 4 карты. 

In [1]:
from math import factorial

def сombination (n, k):
    return int(factorial(n) / (factorial(k) * factorial(n - k)))

Общее число исходов

$$C_n^k = \frac{n!}{k!(n - k)!}$$  
$$C_4^{52} = \frac{52!}{4!(52 - 4)!} = 270725$$

In [2]:
n = сombination(52, 4)
print(f'Общее число исходов: {n}')

Общее число исходов: 270725


a. Найти вероятность того, что все карты – крести. 
$$C_4^{13} = \frac{13!}{4!(13 - 4)!} = 715$$

In [3]:
m = сombination(13, 4)
print(f'Общее число исходов, когда все из 4-х карт крести: {m}')

Общее число исходов, когда все из 4-х карт крести: 715


Вероятность, что все карты крести:  
$$P = \frac{m}{n} = \frac{715}{270725} = 0.00264$$ 

In [4]:
p = m/n
print(f'Вероятность вытащить из колоды 4 карты масти крести: {p}')

Вероятность вытащить из колоды 4 карты масти крести: 0.0026410564225690276


Вероятность вытащить из колоды 4 карты масти крести: 0.00264 или 0.264%.

 
б. Найти вероятность, что среди 4-х карт окажется хотя бы один туз.

In [5]:
m = сombination(4, 1) * сombination(51, 3)
print(f'Количество сочетаний из 4-х карт хотя бы с одним тузом: {m}')

Количество сочетаний из 4-х карт хотя бы с одним тузом: 83300


In [6]:
p =m/n
print(f'Вероятность вытащить четыре карты хотя бы с одним тузом: {p}')

Вероятность вытащить четыре карты хотя бы с одним тузом: 0.3076923076923077


Вероятность вытащить из колоды 4 карты хотя бы с одним тузом: 0.30769 или 30.769%.

2. На входной двери подъезда установлен кодовый замок, содержащий десять кнопок с цифрами от 0 до 9. Код содержит три цифры, которые нужно нажать одновременно. Какова вероятность того, что человек, не знающий код, откроет дверь с первой попытки?

In [7]:
n = сombination(10, 3)
print(f'Количество сочетаний из 3-х цифр от 0 до 9: {n}')

Количество сочетаний из 3-х цифр от 0 до 9: 120


In [8]:
p =1/n
print(f'Вероятность угадать код с первой попытки: {p}')

Вероятность угадать код с первой попытки: 0.008333333333333333


Вероятность угадать код с первой попытки: 0.00833 или 0.833%

3. В ящике имеется 15 деталей, из которых 9 окрашены. Рабочий случайным образом извлекает 3 детали. Какова вероятность того, что все извлеченные детали окрашены?

In [9]:
n = сombination(15, 3)
print(f'Общее число исходов: {n}')

Общее число исходов: 455


In [10]:
m = сombination(9, 3)
print(f'Количество сочетаний из 3-х окрашенных деталей: {m}')

Количество сочетаний из 3-х окрашенных деталей: 84


In [11]:
p = m/n
print(f'Вероятность вытащить три окрашенные детали: {p}')

Вероятность вытащить три окрашенные детали: 0.18461538461538463


Вероятность вытащить три окрашенные детали: 0.18462 или 18.461%

4. В лотерее 100 билетов. Из них 2 выигрышных. Какова вероятность того, что 2 приобретенных билета окажутся выигрышными?

In [12]:
n = сombination(100, 2)
print(f'Общее число исходов, из 2-х билетов: {n}')

Общее число исходов, из 2-х билетов: 4950


In [13]:
m = сombination(2, 2)
print(f'Количество положительных исходов: {m}')

Количество положительных исходов: 1


In [14]:
p = m/n
print(f'Вероятность вытащить два выигрышных билета: {p}')

Вероятность вытащить два выигрышных билета: 0.00020202020202020202


Вероятность угадать код с первой попытки: 0.00020 или 0.020%